fedimint-client/src/transaction/builder.rs

```diff
@@ -13,17 +13,20 @@ use crate::module::StateGenerator;
 use crate::sm::DynState;
 
 #[derive(Clone)]
-pub struct ClientInput<I = DynInput, S = DynState> {
+pub struct ClientInput<I = DynInput> {
     pub input: I,
     pub keys: Vec<KeyPair>,
     pub amount: Amount,
+}
+
+#[derive(Clone)]
+pub struct ClientInputSM<S = DynState> {
     pub state_machines: StateGenerator<S>,
```

Comment:
That's the root of refactoring. Now the state machine generator is not part of the `ClientInput` and is passed independently, so "one state machine driving a group of N inputs" is more natural to express.